Compile error on Gentoo Linux: `no member named 'any_of' in namespace 'std'`

### Problem description

There are two build errors ([4.13.5](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-util/ccache/ccache-4.13.5.ebuild) and master branch) with the following error message:

`error: no member named 'any_of' in namespace 'std'; did you mean '__any_of'?`

### How to reproduce the problem

`emerge =dev-util/cmake-4.13.5`

### Actual behavior

- `src/ccache/storage/storage.cpp`
```bash
[33/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/core/statistics.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/core/statistics.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/core/statistics.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/core/statistics.cpp
[34/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/storage/storage.cpp
FAILED: [code=1] src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o
/usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/storage/storage.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/storage/storage.cpp
/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/storage/storage.cpp:275:14: error: no member named 'any_of' in namespace 'std'; did you mean '__any_of'?
  275 |     if (std::any_of(backward_compat_attributes.begin(),
      |              ^~~~~~
      |              __any_of
/usr/include/c++/v1/__algorithm/simd_utils.h:139:46: note: '__any_of' declared here
  139 | [[__nodiscard__]] _LIBCPP_HIDE_FROM_ABI bool __any_of(__simd_vector<_Tp, _Np> __vec) noexcept {
      |                                              ^
/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/storage/storage.cpp:275:9: error: no matching function for call to '__any_of'
  275 |     if (std::any_of(backward_compat_attributes.begin(),
      |         ^~~~~~~~~~~
/usr/include/c++/v1/__algorithm/simd_utils.h:139:46: note: candidate function template not viable: requires single argument '__vec', but 3 arguments were provided
  139 | [[__nodiscard__]] _LIBCPP_HIDE_FROM_ABI bool __any_of(__simd_vector<_Tp, _Np> __vec) noexcept {
      |                                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
[35/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/config.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/config.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/config.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/config.cpp
[36/74] /usr/lib/llvm/23/bin/clang++  -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -MD -MT src/third_party/fmt/CMakeFiles/dep_fmt.dir/fmt/format.cc.o -MF src/third_party/fmt/CMakeFiles/dep_fmt.dir/fmt/format.cc.o.d -o src/third_party/fmt/CMakeFiles/dep_fmt.dir/fmt/format.cc.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt/fmt/format.cc
[37/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/storage/remote/remotestorage.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/storage/remote/remotestorage.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/storage/remote/remotestorage.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/storage/remote/remotestorage.cpp
```
- `src/ccache/util/path.cpp`
```bash
[45/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/path.cpp
FAILED: [code=1] src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o
/usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/util/path.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/path.cpp
/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/path.cpp:125:8: error: no member named 'sort' in namespace 'std'
  125 |   std::sort(relpath_candidates.begin(),
      |        ^~~~
/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/path.cpp:173:15: error: no member named 'any_of' in namespace 'std'; did you mean '__any_of'?
  173 |   return std::any_of(
      |               ^~~~~~
      |               __any_of
/usr/include/c++/v1/__algorithm/simd_utils.h:139:46: note: '__any_of' declared here
  139 | [[__nodiscard__]] _LIBCPP_HIDE_FROM_ABI bool __any_of(__simd_vector<_Tp, _Np> __vec) noexcept {
      |                                              ^
/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/path.cpp:173:10: error: no matching function for call to '__any_of'
  173 |   return std::any_of(
      |          ^~~~~~~~~~~
/usr/include/c++/v1/__algorithm/simd_utils.h:139:46: note: candidate function template not viable: requires single argument '__vec', but 3 arguments were provided
  139 | [[__nodiscard__]] _LIBCPP_HIDE_FROM_ABI bool __any_of(__simd_vector<_Tp, _Np> __vec) noexcept {
      |                                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
[46/74] /usr/lib/llvm/23/bin/clang++  -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build -I/var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/cxxurl -isystem /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/third_party/fmt  -march=native -O3 -g0 -D_FORTIFY_SOURCE=3 -DNDEBUG -Wno-incompatible-pointer-types -stdlib=libc++ -std=c++20 -include /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999_build/config.h -Wall -Wcast-align -Wdouble-promotion -Wextra -Wextra-semi -Wnon-virtual-dtor -Wnull-dereference -Woverloaded-virtual -Wpedantic -Wshadow -Wunused -Werror -Wextra-semi-stmt -Wno-shadow-field-in-constructor -Wno-gnu-zero-variadic-macro-arguments -Wno-zero-as-null-pointer-constant -Wno-undefined-func-template -Wno-c2y-extensions -MD -MT src/ccache/CMakeFiles/ccache_framework.dir/util/tokenizer.cpp.o -MF src/ccache/CMakeFiles/ccache_framework.dir/util/tokenizer.cpp.o.d -o src/ccache/CMakeFiles/ccache_framework.dir/util/tokenizer.cpp.o -c /var/tmp/portage/dev-util/ccache-9999/work/ccache-9999/src/ccache/util/tokenizer.cpp
```

### Expected behavior

Compiles fine after applying the following patch

<details>

<Summary>ccache-musl.patch</Summary>

```patch
--- a/src/ccache/util/path.cpp
+++ b/src/ccache/util/path.cpp
@@ -16,6 +16,8 @@
 // this program; if not, write to the Free Software Foundation, Inc., 51
 // Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+#include <algorithm>
+
 #include "path.hpp"
 
 #include <ccache/util/direntry.hpp>
--- a/src/ccache/storage/storage.cpp
+++ b/src/ccache/storage/storage.cpp
@@ -48,6 +48,7 @@
 
 #include <cxxurl/url.hpp>
 
+#include <algorithm>
 #include <cmath>
 #include <memory>
 #include <string>

```

</details>

### Ccache version

4.13.5; https://github.com/ccache/ccache/commit/ddd9437

### Ccache configuration

-

### Operating system

Linux

### Environment

OS: Gentoo Linux amd64 musl libc (LLVM profile, libc++)

--- a/src/ccache/storage/storage.cpp
+++ b/src/ccache/storage/storage.cpp
@@ -48,6 +48,7 @@
 
 #include <cxxurl/url.hpp>
 
+#include <algorithm>
 #include <cmath>
 #include <memory>
 #include <string>

--- a/src/ccache/util/path.cpp
+++ b/src/ccache/util/path.cpp
@@ -23,6 +23,8 @@
 #include <ccache/util/format.hpp>
 #include <ccache/util/string.hpp>
 
+#include <algorithm>
+
 #ifdef _WIN32
 const char k_dev_null_path[] = "nul:";
 #else

